core/src/test/java/org/apache/iceberg/TestRewriteManifests.java

```diff
@@ -146,7 +147,7 @@ public void testReplaceManifestsSeparate() {
         .clusterBy(file -> file.path())
         .commit();
 
-    List<ManifestFile> manifests = table.currentSnapshot().allManifests();
+    List<ManifestFile> manifests = new ArrayList(table.currentSnapshot().allManifests());
```

Comment:
Why is this needed?

Reply:
@rdblue sorting isn't supported on immutable lists.

Sorry, I should've closed this earlier. I was thinking this morning that this definitely isn't worth the risk of changing

Reply:
No problem!